src/droidy/droidy-host-session.vala

```diff
@@ -230,17 +230,12 @@ namespace Frida {
 		}
 
 		public async IOStream open_channel (string address, Cancellable? cancellable = null) throws Error, IOError {
-			if (address.has_prefix ("tcp:")) {
-				ulong raw_port;
-				if (!ulong.try_parse (address.substring (4), out raw_port) || raw_port == 0 || raw_port > uint16.MAX)
-					throw new Error.INVALID_ARGUMENT ("Invalid TCP port");
-				uint16 port = (uint16) raw_port;
-
+			if (address.split(":").length == 2) {

```

Comment:
```suggestion
			if (address.contains (":")) {
```

- Missing space.
- Only check that there's at least one ":" as the name of an abstract UNIX domain socket could theoretically contain ":".